### DISCUSSION TITLE
Need Help on Pushing to Github

### DISCUSSION THREAD

**Discussion by XichengSpencer** [oppia/oppia-android#4952] [Q&A (Other)] [answered]
I follow the [guide](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#step-1-making-a-local-code-change) to check out a new branch on top of _develop_ branch and tired to push to my branch. But I get denied here.

Here is the message:"Permission to oppia/oppia-android.git denied to XichengSpencer. unable to access 'https://github.com/oppia/oppia-android.git/': The requested URL returned error: 403" I think since it has my user name on it, I should be logged in already. I don't know if there are other potential reasons.

**Comment by adhiamboperes** [comment, marked as answer]
Hi @XichengSpencer, this is happening because you are trying to push to the main repo instead of a branch on your fork.

Please refer to [documentation](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR) on how to submit a PR and additional references: [Creating a fork](https://app.egghead.io/lessons/javascript-how-to-fork-and-clone-a-github-repository?pl=how-to-contribute-to-an-open-source-project-on-github),[ Contributing to opensource](https://app.egghead.io/playlists/how-to-contribute-to-an-open-source-project-on-github) and Contributing at [Oppia Android](https://github.com/oppia/oppia-android/wiki/Contributing-to-Oppia-android)(Be sure to complete the onboarding steps.)

**Reply by MohitGupta121** [reply]
Thanks @adhiamboperes we fork instructions in our wiki.

@XichengSpencer please follow the instructions in this particular [doc](https://github.com/oppia/oppia-android/wiki/Installing-Oppia-Android#install-oppia-android) to setup and contribute to oppia-android.
If you have any query feel free to ask in github-discussion. 
Thanks

**Comment by XichengSpencer** [comment]
Thank you both for your explanation, I have submitted a PR successfully. The problem is just the I didn't fork before everything.
